sns_ik_lib/src/osns_sm_velocity_ik.cpp

```diff
@@ -0,0 +1,87 @@
+/*! \file osns_sm_velocity_ik.hpp
+ * \brief Optimal SNS, with scale margin, velocity IK solver
+ * \author Fabrizio Flacco
+ * \author Forrest Rogers-Marcovitz
+ */
+/*
+ *    Copyright 2016 Rethink Robotics
+ *    
+ *    Copyright 2012-2016 Fabrizio Flacco
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <sns_ik/osns_sm_velocity_ik.hpp>
+
+using namespace Eigen;
+using namespace sns_ik;
+
+OSNSsmVelocityIK::OSNSsmVelocityIK(int dof, Scalar loop_period) :
```

Comment:
sm == scale margin

Expanding the acronym would expand the file name:
osns_sm_velocity_ik.hpp -> osns_scaleMargin_velocity_ik.hpp

and class name:
OSNSsmVelocityIK -> OSNSScaleMarginVelocityIK

Without context, does the much longer name make things any clearer?